Xenon/Core/XCPU/IIC/IIC.cpp

```diff
@@ -26,79 +26,109 @@ void Xe::XCPU::IIC::XenonIIC::writeInterrupt(u64 intAddress, u64 intData) {
   switch (ppeIntCtrlBlckReg) {
   case Xe::XCPU::IIC::CPU_WHOAMI:
     iicState.ppeIntCtrlBlck[ppeIntCtrlBlckID].REG_CPU_WHOAMI =
-        static_cast<u32>(std::byteswap<u64>(intData));
+      static_cast<u32>(_byteswap_uint64(intData));
     break;
   case Xe::XCPU::IIC::CPU_CURRENT_TSK_PRI:
     iicState.ppeIntCtrlBlck[ppeIntCtrlBlckID].REG_CPU_CURRENT_TSK_PRI =
-        static_cast<u32>(std::byteswap<u64>(intData));
+      static_cast<u32>(_byteswap_uint64(intData));
     break;
   case Xe::XCPU::IIC::CPU_IPI_DISPATCH_0:
     iicState.ppeIntCtrlBlck[ppeIntCtrlBlckID].REG_CPU_IPI_DISPATCH_0 =
-        static_cast<u32>(std::byteswap<u64>(intData));
+      static_cast<u32>(_byteswap_uint64(intData));
     genInterrupt(intType, cpusToInterrupt);
     break;
   case Xe::XCPU::IIC::INT_0x30:
     // Dont know what this does, lets cause an interrupt?
     genInterrupt(intType, cpusToInterrupt);
     break;
   case Xe::XCPU::IIC::EOI:
-    // Check if we have interrupts in the queue.
-    if (!iicState.ppeIntCtrlBlck[ppeIntCtrlBlckID].intQueue.empty()) {
-      // Remove the first interrupt in the Interrupt queue.
-      iicState.ppeIntCtrlBlck[ppeIntCtrlBlckID].intQueue.pop();
-      // Clear the ACK flag.
-      iicState.ppeIntCtrlBlck[ppeIntCtrlBlckID].intAck = false;
+    // If there are interrupts stored in the queue, remove the ack'd.
```

Comment:
Consider making a new temp value using lvalue references to increase readability/performance
`const auto &interrupts = iicState.ppeIntCtrlBlck[ppeIntCtrlBlckID].interrupts;`